File: ACORN Generator/ACORN.cpp
#include <iostream>
#include <vector>
using namespace std;

vector<double> acornGenerator(vector<int> seeds, int m, int N) {
    int k = seeds.size() - 1;
    vector<vector<int>> X(N + 1, vector<int>(k + 1, 0));
    vector<double> U;

    X[0] = seeds;

    for (int n = 1; n <= N; ++n) {
        X[n][0] = X[n - 1][0];
        for (int j = 1; j <= k; ++j) {
            X[n][j] = (X[n - 1][j] + X[n][j - 1]) % m;
        }
        U.push_back(static_cast<double>(X[n][k]) / m);
    }

    return U;
}

int main() {
    vector<int> seeds = {1, 2, 3}; // k = 2
    int m = 10007;
    int N = 10;

    vector<double> randomNumbers = acornGenerator(seeds, m, N);
    cout << "Generated ACORN random numbers:\n";
    for (double num : randomNumbers)
        cout << num << "\n";

    return 0;
}

File: ACORN Generator/ACORN.py
def acorn_generator(seeds, m, N):
    k = len(seeds) - 1  # order
    X = [[0] * (k + 1) for _ in range(N + 1)]
    X[0] = seeds[:]
    U = []

    for n in range(1, N + 1):
        X[n][0] = X[n - 1][0]
        for j in range(1, k + 1):
            X[n][j] = (X[n - 1][j] + X[n][j - 1]) % m
        U.append(X[n][k] / m)
    
    return U

# Example usage
seeds = [1, 2, 3]  # k = 2
m = 10007
N = 10
random_numbers = acorn_generator(seeds, m, N)
print("Generated ACORN random numbers:", random_numbers)
